client/components/form-sections/application-table.tsx

```diff
@@ -1,110 +1,49 @@
-  return (
-    <div>
-      <table className='min-w-full text-center'>
-        <TableHeader />
-        {ApplicationTableData && applications.applications.length != 0 ? (
-          <TableBody applications={applications.applications} />
-        ) : (
-          <p className='text-center text-sm mt-4'>No applications found.</p>
-        )}
-      </table>
-    </div>
-  );
-};
+import { useEffect, useState } from 'react';
+import axios from 'axios';
 
 export const ApplicationDashboard: React.FC<any> = () => {
   const [currentPage, setCurrentPage] = useState<number>(1);
-  const [applicationsPerPage] = useState<number>(10);
-  const [searchValue, setSearchValue]: [string, (search: string) => void] = useState('');
+  const [applicationsPerPage] = useState<number>(2);
+  const [totalApplications, setTotalApplications] = useState<number>();
+  const [searchFacilityName, setSearchFacilityName] = useState<string>('');
+  const [searchConfirmatioID, setSearchConfirmatioID] = useState<string>('');
+  const [data, setData] = useState<any>([]);
 
-  const handleSearch = (e: { target: { value: string } }) => {
-    setSearchValue(e.target.value);
+  const fetchData = async (clearData: boolean) => {
+    let url;
+    if (clearData) {
+      url = `${
+        process.env.NEXT_PUBLIC_SERVER_URL
+      }/applications?facilityName=${''}&confirmationId=${''}&orderBy=status&page=${currentPage}&limit=${applicationsPerPage}`;
+    } else {
+      url = `${process.env.NEXT_PUBLIC_SERVER_URL}/applications?facilityName=${searchFacilityName}&confirmationId=${searchConfirmatioID}&orderBy=status&page=${currentPage}&limit=${applicationsPerPage}`;
+    }
+    const res = await axios.get(url);
+    return res;
   };
 
-  // Filter data with search value
-  const filteredApplications =
-    ApplicationTableData &&
-    ApplicationTableData.filter((item: any) => {
-      return item.Confirmation_ID.toLowerCase().includes(searchValue.toLowerCase());
-    });
+  const setApplicationData = async (clearData: boolean) => {
+    try {
+      const data = await fetchData(clearData);
+      setData(data.data.result);
+      setTotalApplications(data.data.total);
+    } catch (err) {
+      // console.log('Error occured when fetching applications');
+    }
+  };
 
-  // Get current Applications
-  const indexOfLastApplication = currentPage * applicationsPerPage;
-  const indexOfFirstApplication = indexOfLastApplication - applicationsPerPage;
-  const currentApplications = filteredApplications.slice(
-    indexOfFirstApplication,
-    indexOfLastApplication,
-  );
+  useEffect(() => {
+    (async () => {
+      setApplicationData(false);
+    })();
+  }, [currentPage]);
 
   // Change pages
   const nextPage = () => {
-    if (currentPage == Math.floor(filteredApplications.length / applicationsPerPage) + 1) return;
+    if (!totalApplications || currentPage == Math.ceil(totalApplications / applicationsPerPage))
+      return;
     setCurrentPage(currentPage + 1);
```

Comment:
We shouldn't update the state of this. We should only update the browser URL query param and the state change should be captured using an useEffect and this must be handled!